metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2908,17 +2928,24 @@ def emit_dashboards(self) -> Iterable[MetadataWorkUnit]:
             c.DASHBOARDS_CONNECTION,
             dashboards_filter,
         ):
-            yield from self.emit_dashboard(dashboard, dashboard.get(c.WORKBOOK))
+            # LUID is blank if the dashboard is hidden in the workbook.
+            # More info here: https://help.tableau.com/current/api/metadata_api/en-us/reference/dashboard.doc.html
+            if self.config.ingest_hidden_assets or dashboard.get(c.LUID):
```

Comment:
My two cents: as the Tableau logic for hidden dashboards is quite "befogged", maybe it could make sense to add a `def is_hidden_workbook(self, dashboard):  return self.config.ingest_hidden_assets or dashboard.get(c.LUID)` just to avoid to replicate the block (your comments included) several times in the file. Perhaps, It could help future maintenance.